great_expectations/profile/json_schema_profiler.py

```diff
@@ -0,0 +1,253 @@
+        result = super().validate(schema)
+        if not result:
+            return False
+
+        # TODO validate that the dict is a valid JSONSchema document. Pretty meta huh!
+        #  Note this will add a dependency.
+        return True
+
+    def _profile(self, schema: dict, suite_name: Union[dict, str]) -> ExpectationSuite:
+        expectations = []
+        if schema["type"] == JsonSchemaTypes.OBJECT.value:
+            for key, details in schema["properties"].items():
+                # TODO configuration switches to affect types?
+                expectations.append(self._create_existence_expectation(key, details))
+
+                type_expectation = self._create_type_expectation(key, details)
+                if type_expectation:
+                    expectations.append(type_expectation)
+
+                range_expectation = self._create_range_expectation(key, details)
+                if range_expectation:
+                    expectations.append(range_expectation)
+
+                boolean_expectation = self._create_boolean_expectation(key, details)
+                if boolean_expectation:
+                    expectations.append(boolean_expectation)
+
+                set_expectation = self._create_set_expectation(key, details)
+                if set_expectation:
+                    expectations.append(set_expectation)
+
+                string_len_expectation = self._create_string_length_expectation(
+                    key, details
+                )
+                if string_len_expectation:
+                    expectations.append(string_len_expectation)
+        description = schema.get("description", None)
+        meta = None
+        if description:
+            meta = {
+                "notes": {
+                    "format": "markdown",
+                    "content": [f"### Description:\n{description}"],
+                }
+            }
+        suite = ExpectationSuite(suite_name, expectations=expectations, meta=meta)
+        suite.add_citation(
+            comment=f"This suite was built by the {self.__class__.__name__}",
+        )
+        return suite
+
+    def _create_existence_expectation(
+        self, key: str, details: dict
+    ) -> ExpectationConfiguration:
+        kwargs = ExpectationKwargs(column=key)
+        description = details.get("description", None)
+        meta = None
+        if description:
+            meta = {
+                "notes": {
+                    "format": "markdown",
+                    "content": [f"### Description:\n{description}"],
+                }
+            }
+        return ExpectationConfiguration("expect_column_to_exist", kwargs, meta=meta)
+
+    def _create_type_expectation(
+        self, key: str, details: dict
+    ) -> Optional[ExpectationConfiguration]:
+        type_ = details.get("type", None)
+        if type_ is None:
+            return None
+
+        # TODO map JSONSchema types to which type backend? Pandas?
```

Comment:
In this case we need the actual mapping for json schema types to the types of supported backends similarly to the way that the BasicDatasetProfilerBase contains predefined sets of type names; I think using those is the best bet, for example:

```
if type_ == 'integer':
  type_expectation = BasicDatasetProfilerBase.INT_TYPE_NAMES
```

Reply:
I extracted those type mappings into a separate class because I didn't want coupling between two peer classes. However, we may need to pair on this because I don't yet see your vision here.

Reply:
Let's do a quick sync...I just want to ensure there's an (availalbe) mapping from the JSONSchema-specific verbiage to backend-specific verbiage.